algorithms/kernel/classifier/classifier_predict_fpt.cpp

```diff
@@ -45,13 +48,47 @@ template <typename algorithmFPType>
 DAAL_EXPORT services::Status Result::allocate(const daal::algorithms::Input *input, const daal::algorithms::Parameter *parameter, const int method)
 {
     services::Status st;
-    set(prediction, data_management::HomogenNumericTable<algorithmFPType>::create(1, (static_cast<const InputIface *>(input))->getNumberOfRows(), data_management::NumericTableIface::doAllocate, &st));
+    set(prediction, HomogenNumericTable<algorithmFPType>::create(1, (static_cast<const InputIface *>(input))->getNumberOfRows(), NumericTableIface::doAllocate, &st));
     return st;
 }
 
 template DAAL_EXPORT services::Status Result::allocate<DAAL_FPTYPE>(const daal::algorithms::Input *input, const daal::algorithms::Parameter *par, const int method);
 
 }
+namespace interface2
+{
+
+/**
+ * Allocates memory for storing prediction results of the classification algorithm
+ * \tparam  algorithmFPType     Data type for storing prediction results
+ * \param[in] input     Pointer to the input objects of the classification algorithm
+ * \param[in] parameter Pointer to the parameters of the classification algorithm
+ * \param[in] method    Computation method
+ */
+template <typename algorithmFPType>
+DAAL_EXPORT services::Status Result::allocate(const daal::algorithms::Input *input, const daal::algorithms::Parameter *parameter, const int method)
+{
+    services::Status st;
+    const Parameter *par = static_cast<const Parameter *>(parameter);
+    DAAL_CHECK(par, services::ErrorNullParameterNotSupported);
+
+    const size_t nRows = (static_cast<const InputIface *>(input))->getNumberOfRows();
+    const size_t nClasses = par->nClasses;
+
+    if(par->resultsToEvaluate & computeClassesLabels)
+        set(prediction, HomogenNumericTable<algorithmFPType>::create(1, nRows, NumericTableIface::doAllocate, &st));
```

Comment:
Why don`t you use create method with default value as bellow?

Reply:
will fix